nncf/experimental/quantization/quantizers/openvino_quantizer.py

```diff
@@ -0,0 +1,285 @@
+            weights_quantization_params=weights_quantization_params,
+            quantizer_propagation_rule=quantizer_propagation_rule,
+        )
+
+    def get_quantization_setup(self, model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> SingleConfigQuantizerSetup:
+        self._min_max_algo._set_backend_entity(model)
+        return self._min_max_algo.find_quantization_setup(model, nncf_graph)
+
+    def annotate(self, model: torch.fx.GraphModule) -> torch.fx.GraphModule:
+        nncf_graph = GraphConverter.create_nncf_graph(model)
+        quantization_setup = self.get_quantization_setup(model, nncf_graph)
+
+        non_unified_quantization_points = quantization_setup.quantization_points.copy()
+
+        graph = model.graph
+        node_vs_torch_annotation = defaultdict(TorchAOQuantizationAnnotation)
+        for quantizer_ids in quantization_setup.unified_scale_groups.values():
+
+            root_quantizer_id = self._get_unified_scales_root_quantizer_id(
+                nncf_graph, quantizer_ids, quantization_setup
+            )
+            qp = quantization_setup.quantization_points[root_quantizer_id]
+            root_edge_or_node, annotation = self._get_edge_or_node_and_annotation(graph, qp, node_vs_torch_annotation)
+            qspec = self._get_inductor_qspec_from_qp(qp)
+            self._fill_torch_ao_annotation(root_edge_or_node, qspec, annotation)
+
+            while quantizer_ids:
+                quantizer_id = quantizer_ids.pop()
+                del non_unified_quantization_points[quantizer_id]
+
+                if quantizer_id == root_quantizer_id:
+                    continue
+
+                qspec = TorchAOSharedQuantizationSpec(root_edge_or_node)
+                qp = quantization_setup.quantization_points[quantizer_id]
+                edge_or_node, annotation = self._get_edge_or_node_and_annotation(graph, qp, node_vs_torch_annotation)
+
+                self._fill_torch_ao_annotation(edge_or_node, qspec, annotation)
+
+        for qp in non_unified_quantization_points.values():
+            edge_or_node, annotation = self._get_edge_or_node_and_annotation(graph, qp, node_vs_torch_annotation)
+            qspec = self._get_inductor_qspec_from_qp(qp)
+            self._fill_torch_ao_annotation(edge_or_node, qspec, annotation)
+
+        for node, annotation in node_vs_torch_annotation.items():
+            assert QUANT_ANNOTATION_KEY not in node.meta
+            node.meta[QUANT_ANNOTATION_KEY] = annotation
+
+    @staticmethod
+    def _get_unified_scales_root_quantizer_id(
+        nncf_graph: NNCFGraph, quantizer_ids: List[int], quantizer_setup: SingleConfigQuantizerSetup
+    ) -> int:
+        """
+        Identifies the earliest quantizer node ID based on the corresponding `nncf_node.node_id`
+        in the given NNCFGraph. This is required by the `_get_obs_or_fq_map` function.
+        Refer to: https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/pt2e/prepare.py#L291
```

Comment:
Honestly, it is not clear from the provided link why the earliest quantizer is needed. Let's discuss it offline.